drivers/storage/portworx/component/portworx_basic.go

```diff
@@ -521,6 +537,118 @@ func getPortworxServiceSpec(
 	return newService
 }
 
+func (c *portworxBasic) createAndMaintainPxSaTokenSecret(cluster *corev1.StorageCluster, ownerRef *metav1.OwnerReference) error {
+	if err := c.createTokenSecretIfNotExist(cluster, ownerRef); err != nil {
+		return err
+	}
+	if err := c.maintainTokenSecret(cluster, ownerRef); err != nil {
+		return fmt.Errorf("failed to maintain the token secret for px container. %v", err)
+	}
+	return nil
+}
+
+func (c *portworxBasic) createTokenSecretIfNotExist(cluster *corev1.StorageCluster, ownerRef *metav1.OwnerReference) error {
+	secret := &v1.Secret{}
+	err := c.k8sClient.Get(context.TODO(),
+		types.NamespacedName{
+			Name:      pxutil.PortworxServiceAccountTokenSecretName,
+			Namespace: cluster.Namespace,
+		}, secret)
+	if err != nil {
+		if errors.IsNotFound(err) || len(secret.Data) == 0 || len(secret.Data[v1.ServiceAccountTokenKey]) == 0 {
+			token, err := generateToken(cluster)
+			if err != nil {
+				return err
+			}
+			rootCaCrt, _ := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/ca.crt")
```

Comment:
don't ignore error

Reply:
How shall we handle the error if the file doesn't exist? If it's in an actual k8s cluster, the file is supposed to be there because it's part of the service account secret. But it doesn't exist in UTs and breaking all the tests

Reply:
If the file does not exist, we don't need to add it to the token. But what if it is some other error? Should we handle that?